src/top/state.rs

```diff
@@ -32,6 +32,8 @@ pub enum EventType {
     /// Interval + identified metric
     ProcessedBytesThroughputs(i64, Vec<IdentifiedMetric>),
     ErrorsTotals(Vec<IdentifiedMetric>),
+    #[cfg(feature = "allocation-tracing")]
```

Comment:
This is _likely_ to be fine as users will typically be using the same `vector` binary to run `vector top` as is serving the API, but there is a possibility for this not to be the case. That is it is possible that the user is running `vector top` with a Vector that was compiled without allocation tracking where the `vector` serving the API, was, or vice versa. I think a better way to detect this would be for the API to return whether allocation tracking is enabled or not.

Reply:
Gotcha, yeah I see. For `A is running vector top, and B is the server` I'm wondering if we could/should write a warning message somewhere so users know why they aren't seeing memory use information? I know `vector top` uses the whole terminal so we can't just write to stderr but maybe we could flash the warning in the bottom box where it says "To quit..."?

Reply:
Not blocking though, just if it is easy to do.

Reply:
I believe we should do this in a follow up PR.

Reply:
Also am thinking of doing some refactoring in the `vector top` section of the codebase.

Reply:
Sounds good to me 👍